app/SignUp/page.js

```diff
@@ -0,0 +1,153 @@
+"use client";
+import { useRef } from "react";
+import { useState } from "react";
+import { Box, TextField, Button, Typography, Input } from "@mui/material";
+import { Label } from "@mui/icons-material";
+import { useRouter } from "next/navigation";
+
+export default function SignUp() {
+  const router = useRouter();
+
+  const [emailError, setEmailError] = useState(false);
+  const [emailErrorMessage, setEmailErrorMessage] = useState("");
+  const [passwordError, setPasswordError] = useState(false);
+  const [passwordErrorMessage, setPasswordErrorMessage] = useState("");
+  const [firstNameError, setFirstNameError] = useState(false);
+  const [firstNameErrorMessage, setFirstNameErrorMessage] = useState("");
+  const [lastNameError, setLastNameError] = useState(false);
+  const [lastNameErrorMessage, setLastNameErrorMessage] = useState("");
+
+  const firstNameRef = useRef(null);
+  const lastNameRef = useRef(null);
+  const emailRef = useRef(null);
+  const passwordRef = useRef(null);
+
+  const handleKeyDown = (event, nextFieldRef) => {
+    if (event.key === "Enter") {
+      event.preventDefault();
+      nextFieldRef.current?.focus();
+    }
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const isValid = validateInputs();
+    if (isValid) {
+      alert("Your form has been submitted");
+      router.push("/");
+    }
+  };
+
+  const validateInputs = () => {
+    const email = document.getElementById("email");
+    const password = document.getElementById("password");
+    const firstName = document.getElementById("first-name");
```

Comment:
We are asked to use refs to implement validation that way. Look at how we did it in class. You are not to manipulate the DOM directly like this in React.

Reply:
I used the useRef for the validation instead of DOM manipulation